packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -229,35 +224,46 @@ library LibProposing {
         // verification.
         // Prover can charge ERC20/NFT as fees; msg.value can be zero. Taiko
         // doesn't mandate Ether as the only proofing fee.
-        blk.assignedProver = params.assignment.prover;
-
-        // The assigned prover burns Taiko tokens, referred to as the
-        // "liveness bond." This bond remains non-refundable to the
-        // assigned prover under two conditions: if the block's verification
-        // transition is not the initial one or if it was generated and
-        // validated by different provers. Instead, a portion of the assignment
-        // bond serves as a reward for the actual prover.
-        LibTaikoToken.debitTaikoToken(state, resolver, blk.assignedProver, config.livenessBond);
+        blk.assignedProver = params.assignedProver;
 
         // Increment the counter (cursor) by 1.
         unchecked {
             ++state.slotB.numBlocks;
         }
 
-        // Validate the prover assignment, then charge Ether or ERC20 as the
-        // prover fee based on the block's minTier.
-        uint256 proverFee = _payProverFeeAndTip(
-            meta.minTier, meta.blobHash, blk.blockId, blk.metaHash, params.assignment
-        );
-
         emit BlockProposed({
             blockId: blk.blockId,
             assignedProver: blk.assignedProver,
             livenessBond: config.livenessBond,
-            proverFee: proverFee,
             meta: meta,
             depositsProcessed: depositsProcessed
         });
+
+        {
+            TaikoToken tko = TaikoToken(resolver.resolve("taiko_token", false));
+            uint256 tkoBalance = tko.balanceOf(address(this));
+
+            // Run all hooks.
+            // Note that address(this).balance has been updated with msg.value,
+            // prior to any code in this function has been executed.
+            uint256 ethBalance = address(this).balance - msg.value;
```

Comment:
I think no ETH is stored in this contract, so this could simply be `address(this).balance`? Same thing below I think where instead of having to keep track of `msgValue`, it could always just be  `address(this).balance`.

Reply:
True.  I added some comments and removed `ethBalance`